test/fail_compilation/fail21868b.d

```diff
@@ -0,0 +1,22 @@
+/* REQUIRED_ARGS: -preview=dip1000
+TEST_OUTPUT:
+---
+fail_compilation/fail21868b.d(19): Error: returning `&s.x` escapes a reference to parameter `s`
+fail_compilation/fail21868b.d(19):        perhaps annotate the parameter with `return`
```

Comment:
This error message isn't helpful because the parameter is already marked as `return`. It should rather suggest to remove `scope`(?)

Reply:
Yes, but improving error messages is not what this PR is about. Should be a separate PR. This one is solely focused on Issue 21868.

Reply:
Sure, but this breaks existing code without providing an actual help for the user. It's actually misleading them...

Reply:
Ok. It makes things more complex, but not that much.,